### PR TITLE
Return Ok(None) instead of error on metal error from acquire_texture

### PR DESCRIPTION
**Pull request by xacrimon**
**Connections**
https://github.com/gfx-rs/wgpu/pull/1938

### CODE REVIEW

File: wgpu-hal/src/metal/surface.rs [hunk 1]
```diff
@@ -267,38 +267,38 @@ impl crate::Surface<super::Api> for super::Surface {
...
-                },
-            },
-            drawable,
-            present_with_transaction: self.present_with_transaction,
-        };
+        let pair = autoreleasepool(|| {
+            render_layer
+                .next_drawable()
+                .map(|drawable| (drawable.to_owned(), drawable.texture().to_owned()))
+        });
```
Comment (kvark) [inline wgpu-hal/src/metal/surface.rs:274]: you can just add `?` here

Comment (kvark) [reply wgpu-hal/src/metal/surface.rs:274]: I did 4f677c8b67f07d4c13618b9978c0ba7ee4e237bf

**Comment by xacrimon** [comment]
@kvark fixed

**Review by kvark — APPROVED** [review]

**Comment by kvark** [comment]
published in wgpu-hal-0.10.5

**Pull request merged**